pkg/snmp/snmp.go

```diff
@@ -118,8 +118,17 @@ func NewListenerConfig() (ListenerConfig, error) {
 	// Set defaults before unmarshalling
 	snmpConfig.CollectDeviceMetadata = true
 	snmpConfig.CollectTopology = true
-	if err := coreconfig.Datadog.UnmarshalKey("snmp_listener", &snmpConfig, opt); err != nil {
-		return snmpConfig, err
+
+	if coreconfig.Datadog.IsSet("network_devices.snmp_listener") {
+		if err := coreconfig.Datadog.UnmarshalKey("network_devices.snmp_listener", &snmpConfig, opt); err != nil {
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** ‏I personally think that inlining error checks makes the code hard to read, what do you think of the following?
```
err := coreconfig.Datadog.UnmarshalKey("network_devices.snmp_listener", &snmpConfig, opt)
if err != nil {
  return snmpConfig, err
}
```

Reply:
agree with you, https://github.com/DataDog/datadog-agent/pull/24644/commits/4b7dcdfc2e1ab610ba6faaf6056e6e8f53c851c5